### PR TITLE
solve triangular

### PR DESCRIPTION
**Pull request by anurag629**
solve_triangular #10423

### CODE REVIEW

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py [hunk 1]
```diff
@@ -583,3 +583,39 @@ def test_cond(
...
+    dtype_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=2,
+        max_num_dims=2,
+        min_dim_size=2,
+        max_dim_size=10,
+        num_arrays=2,
+        allow_nan=False,
+        shared_dtype=True,
+    ),
```
Comment (Aarsh2001) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py:599]: hey @anurag629, small change. Please generate float values for `lower`, `transpose`, and `unit_diagonal`

File: ivy/functional/backends/torch/experimental/linear_algebra.py [hunk 2]
```diff
@@ -177,3 +178,27 @@ def cond(
...
+
+@with_supported_dtypes({"1.13.0 and below": ("float32", "float64", "complex64", "complex128")}, backend_version)
+def solve_triangular(
+    a: torch.Tensor,
+    b: torch.Tensor,
+    /,
+    *,
+    upper: bool = False,
+    left: bool = False,
+    unitriangular: bool = False,
```
Comment (Aarsh2001) [inline ivy/functional/backends/torch/experimental/linear_algebra.py:191]: These are not the arguments you are returning from the `Ivy API`. Please make sure that the arguments are consistent across backends.

File: ivy/functional/backends/tensorflow/experimental/linear_algebra.py [hunk 3]
```diff
@@ -189,3 +189,24 @@ def cond(
...
+
+
+def solve_triangular(
+    a: Union[tf.Tensor, tf.Variable],
+    b: Union[tf.Tensor, tf.Variable],
+    /,
+    *,
+    lower: bool = True,
+    transpose: bool = False,
+    conjugate: bool = False,
```
Comment (Aarsh2001) [inline ivy/functional/backends/tensorflow/experimental/linear_algebra.py:201]: You should either handle the `conjugate` argument in all backends or skip it completely.

File: ivy/functional/backends/numpy/experimental/linear_algebra.py [hunk 4]
```diff
@@ -167,3 +167,37 @@ def cond(
...
 cond.support_native_out = False
+
+
+def solve_triangular(
+    a: np.ndarray,
+    b: np.ndarray,
+    /,
+    *,
+    lower: bool = True,
+    trans: bool = False,
```
Comment (Aarsh2001) [inline ivy/functional/backends/numpy/experimental/linear_algebra.py:178]: There is no such thing as `trans` being returned from the `Ivy API`. You are also returning `unit_diagonal` in the `Ivy API`, how are you handling it here?

File: ivy/functional/backends/jax/experimental/linear_algebra.py [hunk 5]
```diff
@@ -148,3 +148,15 @@ def cond(
...
     return jnp.linalg.cond(x, p=p)
+
+
+def solve_triangular(
+    a: JaxArray,
+    b: JaxArray,
+    /,
+    *,
+    lower: bool = True,
+    unit_diagonal: bool = False,
```
Comment (Aarsh2001) [inline ivy/functional/backends/jax/experimental/linear_algebra.py:159]: You are also returning `transpose` from the `Ivy API`. How are handling it here ??

File: ivy/functional/backends/torch/experimental/linear_algebra.py (hunk 2 above)
Comment (Aarsh2001) [reply ivy/functional/backends/torch/experimental/linear_algebra.py:191]: For example, where are the arguments `lower`, `unit_diagonal` and `transpose`.

**Review by Aarsh2001 — CHANGES_REQUESTED** [review]
Kindly look at the changes below, thanks.

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py [hunk 6]
```diff
@@ -589,6 +589,67 @@ def test_cond(
...
+            dtype=input_dtype,
+            min_value=-10,
+            max_value=10,
+        )
+    )
+
+    lower = draw(st.booleans())
+    transpose = draw(st.booleans())
+    unit_diagonal = draw(st.booleans())
+
```
Comment (Aarsh2001) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py:620]: It will be better if we do this in `handle_test` decorator 😅.

**Review by Aarsh2001 — CHANGES_REQUESTED** [review]
Hey @anurag629 all looks good, just a small change ,thanks

**Comment by anurag629** [comment]
Hey @Aarsh2001 please review this PR 😃

File: ivy/data_classes/container/experimental/linear_algebra.py [hunk 7]
```diff
@@ -886,6 +886,137 @@ def cond(
...
+    @staticmethod
+    def static_solve_triangular(
+        self: ivy.Container,
+        /,
+        *,
+        key_chains: Optional[Union[List[str], Dict[str, str]]] = None,
+        to_apply: bool = True,
+        prune_unapplied: bool = False,
+        map_sequences: bool = False,
+        b: Union[ivy.Container, ivy.Array, ivy.NativeArray],
```
Comment (Aarsh2001) [inline ivy/data_classes/container/experimental/linear_algebra.py:898]: This is a positional argument, should come before line `892`

File: ivy/data_classes/container/experimental/linear_algebra.py [hunk 8]
```diff
@@ -886,6 +886,137 @@ def cond(
...
+    
+    def solve_triangular(
+        self: ivy.Container,
+        /,
+        *,
+        key_chains: Optional[Union[List[str], Dict[str, str]]] = None,
+        to_apply: bool = True,
+        prune_unapplied: bool = False,
+        map_sequences: bool = False,
+        b: Union[ivy.Container, ivy.Array, ivy.NativeArray],
```
Comment (Aarsh2001) [inline ivy/data_classes/container/experimental/linear_algebra.py:957]: Same here

**Review by Aarsh2001 — CHANGES_REQUESTED** [review]

**Comment by anurag629** [comment]
Hey @Aarsh2001, why these tests are failing? :)

File: ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py [hunk 9]
```diff
@@ -589,6 +589,69 @@ def test_cond(
...
+    matrix_2 = draw(
+        helpers.dtype_and_values(
+            shape=shape_1,
+            dtype=input_dtype,
+            min_value=-10,
+            max_value=10,
+        )
+    )
+
+    return input_dtype, [matrix_1[1][0], matrix_2[1][0]]
```
Comment (Aarsh2001) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py:617]: I feel we don't need this function here, you can simply use `dtype_and_values` and pass `num_arrays=2`. You can also use these keyword arguments for generating the shape values - 

``` 
min_num_dims
max_num_dims
min_dim_size
max_dim_size
```

**Review by Aarsh2001 — CHANGES_REQUESTED** [review]

**Review by Aarsh2001 — CHANGES_REQUESTED** [review]
hi @anurag, if you look at the test logs, you'll see that the value test fails for majority backends. Meaning, your implementation may not be consistent across backends and the ground truth backend tensorflow. I suggest you try to create a demo file in your local machine and run the failing example for all backends. You should verify that all the outputs match.

**Comment by MahmoudAshraf97** [comment]
Hi, @anurag629 are you still working on this?

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
